DNN Platform/Modules/Journal/ServicesController.cs

```diff
@@ -98,135 +98,157 @@ private static bool IsAllowedLink(string url)
+                    var file = FileManager.Instance.GetFile(fileId);
+
+                    if (!IsCurrentUserFile(file))
                     {
-                        ji.ItemData.ImageUrl = string.Empty;
+                        throw new ArgumentException("you have no permission to attach files not belongs to you.");
                     }
 
-                    ji.ItemData.Description = HttpUtility.UrlDecode(ji.ItemData.Description);
+                    ji.ItemData.Title = file.FileName;
+                    ji.ItemData.Url = Globals.LinkClick(ji.ItemData.Url, Null.NullInteger, Null.NullInteger);
 
-                    if (!IsAllowedLink(ji.ItemData.Url))
+                    if (string.IsNullOrEmpty(ji.ItemData.ImageUrl) &&
+                        originalImageUrl.ToLower().StartsWith("/linkclick.aspx?") &&
+                        AcceptedFileExtensions.Contains(file.Extension.ToLower()))
                     {
-                        ji.ItemData.Url = string.Empty;
+                        ji.ItemData.ImageUrl = originalImageUrl;
                     }
+                }
+            }
 
-                    if (!string.IsNullOrEmpty(ji.ItemData.Url) && ji.ItemData.Url.StartsWith("fileid="))
-                    {
-                        var fileId = Convert.ToInt32(ji.ItemData.Url.Replace("fileid=", string.Empty).Trim());
-                        var file = FileManager.Instance.GetFile(fileId);
-
-                        if (!IsCurrentUserFile(file))
-                        {
-                            throw new ArgumentException("you have no permission to attach files not belongs to you.");
-                        }
-
-                        ji.ItemData.Title = file.FileName;
-                        ji.ItemData.Url = Globals.LinkClick(ji.ItemData.Url, Null.NullInteger, Null.NullInteger);
-
-                        if (string.IsNullOrEmpty(ji.ItemData.ImageUrl) &&
-                            originalImageUrl.ToLower().StartsWith("/linkclick.aspx?") &&
-                            AcceptedFileExtensions.Contains(file.Extension.ToLower()))
-                        {
-                            ji.ItemData.ImageUrl = originalImageUrl;
-                        }
-                    }
+            return ji;
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+		[DnnAuthorize(DenyRoles = "Unverified Users")]
+        public HttpResponseMessage Create(CreateDTO postData)
+        {
+            try
+            {
+                int UserID = UserInfo.UserID;
+                IDictionary<string, UserInfo> mentionedUsers = new Dictionary<string, UserInfo>();
+
+                if (postData.ProfileId == -1)
+                {
+                    postData.ProfileId = UserID;
                 }
 
+                checkProfileAccess(postData.ProfileId, UserInfo);
+
+                checkGroupAccess(ref postData);
+
+                // JI
+                var ji = prepareJournalItem(ref postData, ref mentionedUsers);
```

Comment:
By the way, while you're cleaning up maybe you could actually rename that `ji` into `journalItem`

Reply:
done